### ISSUE TITLE
Pioneer occasionally crashing after toggling HUD off in flight-ui

### ISSUE THREAD

**Issue by cwyss** [pioneerspacesim/pioneer#4812]
### Observed behaviour
After toggling HUD off completely in flight-ui (2xTAB) and then back on again (TAB again), pioneer occasionally crashes with a segmentation fault -- sooner or later; though sometimes it does not crash at all.

### Steps to reproduce
This is the tricky part. I managed to find a (rather recent) version of pioneer, with which the crash can be reproduced quite reliably: 105d06e33 (31 Jan 2020)
With this version:

1. Start on Mars
2. Toggle HUD off/on (3xTAB)
3. Undock, retract gear; in system overview window, choose autopilot fly to vicinity of Mariasuriru
4. Speed up time, switch views (fwd, bkwd, internal, external), enjoy the flight :smile: 
5. Upon reaching your target, tell autopilot to dock at Mariasuriru.

For me, the game crashes anywhere in step 3, 4 or 5, sometimes earlier, sometimes later. There is no crash if I do not toggle the HUD off.

I also have a save file [sgf_mariasuriru.gz](https://github.com/pioneerspacesim/pioneer/files/4241928/sgf_mariasuriru.gz) with the ship docked at Mariasuriru, which triggers the crash in a simple way:

1. Load game "sgf_mariasuriru"
2. Unpause
3. Toggle HUD off/on
4. Press F4 -> segmentation fault

Here is the backtrace corresponding to the segfault:
<details>

```
Thread 1 "pioneer" received signal SIGSEGV, Segmentation fault.
ImGui::RenderMouseCursor (draw_list=draw_list@entry=0x5555571fbdd8, pos=..., scale=1, 
    mouse_cursor=0, col_fill=col_fill@entry=4294967295, 
    col_border=col_border@entry=4278190080, col_shadow=col_shadow@entry=805306368)
    at /home/cwyss/games/pioneer/pioneer/contrib/imgui/imgui_draw.cpp:3058
3058	    if (font_atlas->GetMouseCursorTexData(mouse_cursor, &offset, &size, &uv[0], &uv[2]))
(gdb) backtrace
#0  ImGui::RenderMouseCursor (draw_list=draw_list@entry=0x5555571fbdd8, pos=..., scale=1, 
    mouse_cursor=0, col_fill=col_fill@entry=4294967295, 
    col_border=col_border@entry=4278190080, col_shadow=col_shadow@entry=805306368)
    at /home/cwyss/games/pioneer/pioneer/contrib/imgui/imgui_draw.cpp:3058
#1  0x0000555555930672 in ImGui::Render ()
    at /home/cwyss/games/pioneer/pioneer/contrib/imgui/imgui.cpp:4109
#2  0x00005555555eebc9 in PiGui::RenderImGui ()
    at /home/cwyss/games/pioneer/pioneer/src/PiGui.cpp:447
#3  0x00005555555dab4b in Pi::DrawPiGui (delta=<optimized out>, handler="GAME")
    at /home/cwyss/games/pioneer/pioneer/src/Pi.cpp:1551
#4  0x00005555555db84f in Pi::MainLoop ()
    at /home/cwyss/games/pioneer/pioneer/src/Pi.cpp:1405
#5  0x00005555555dc393 in Pi::Start (startPath=...)
    at /home/cwyss/games/pioneer/pioneer/src/Pi.cpp:1225
#6  0x00005555555b9cf8 in main (argc=<optimized out>, argv=<optimized out>)
    at /home/cwyss/games/pioneer/pioneer/src/main.cpp:181
```
</details>

With the first way to get the segfault, starting from Mars, the traceback sometimes looks a little different:
<details>

```
Thread 1 "pioneer" received signal SIGSEGV, Segmentation fault.
ImFontAtlas::GetMouseCursorTexData (this=this@entry=0x7fffec0accb0, 
    cursor_type=<optimized out>, out_offset=out_offset@entry=0x7fffffffd3b0, 
    out_size=out_size@entry=0x7fffffffd3b8, 
    out_uv_border=out_uv_border@entry=0x7fffffffd3d0, 
    out_uv_fill=out_uv_fill@entry=0x7fffffffd3e0)
    at /home/cwyss/games/pioneer/pioneer/contrib/imgui/imgui_draw.cpp:1774
1774	    ImVec2 pos = FONT_ATLAS_DEFAULT_TEX_CURSOR_DATA[cursor_type][0] + ImVec2((float)r.X, (float)r.Y);
(gdb) backtrace
#0  ImFontAtlas::GetMouseCursorTexData (this=this@entry=0x7fffec0accb0, 
    cursor_type=<optimized out>, out_offset=out_offset@entry=0x7fffffffd3b0, 
    out_size=out_size@entry=0x7fffffffd3b8, 
    out_uv_border=out_uv_border@entry=0x7fffffffd3d0, 
    out_uv_fill=out_uv_fill@entry=0x7fffffffd3e0)
    at /home/cwyss/games/pioneer/pioneer/contrib/imgui/imgui_draw.cpp:1774
#1  0x000055555594b2c7 in ImFontAtlas::GetMouseCursorTexData (out_uv_fill=0x7fffffffd3e0, 
    out_uv_border=0x7fffffffd3d0, out_size=0x7fffffffd3b8, out_offset=0x7fffffffd3b0, 
    cursor_type=<optimized out>, this=0x7fffec0accb0)
    at /home/cwyss/games/pioneer/pioneer/contrib/imgui/imgui_draw.cpp:1768
#2  ImGui::RenderMouseCursor (draw_list=draw_list@entry=0x5555571fafd8, pos=..., scale=1, 
    mouse_cursor=<optimized out>, col_fill=col_fill@entry=4294967295, 
    col_border=col_border@entry=4278190080, col_shadow=col_shadow@entry=805306368)
    at /home/cwyss/games/pioneer/pioneer/contrib/imgui/imgui_draw.cpp:3058
#3  0x0000555555930672 in ImGui::Render ()
    at /home/cwyss/games/pioneer/pioneer/contrib/imgui/imgui.cpp:4109
#4  0x00005555555eebc9 in PiGui::RenderImGui ()
    at /home/cwyss/games/pioneer/pioneer/src/PiGui.cpp:447
#5  0x00005555555dab4b in Pi::DrawPiGui (delta=<optimized out>, handler="GAME")
    at /home/cwyss/games/pioneer/pioneer/src/Pi.cpp:1551
#6  0x00005555555db84f in Pi::MainLoop ()
    at /home/cwyss/games/pioneer/pioneer/src/Pi.cpp:1405
#7  0x00005555555dc393 in Pi::Start (startPath=...)
    at /home/cwyss/games/pioneer/pioneer/src/Pi.cpp:1225
#8  0x00005555555b9cf8 in main (argc=<optimized out>, argv=<optimized out>)
    at /home/cwyss/games/pioneer/pioneer/src/main.cpp:181
```
</details>

I also tried these versions of pioneer: 20200203 (release), e255171 (22 Feb 2020)
Here the situation is much more random: Often I do not get any crashes at all using the above steps. Nevertheless, eventually I do got crashes also with these versions. The backtraces are the same then.

### A related bug?

From my observation, the HUD off/on toggling is necessary for the crash to happen. As was reported in #4762, toggling the HUD visibility changes the mouse cursor from the custom pioneer style to the X/Dektop environment's default cursor. This might be related to the traceback, which shows that the segfault happens in imgui code related to rendering the mouse cursor..!

__My pioneer version (and OS):__
Debian 9.12 with Gnome on X11
Pioneer gfx config: 1920x1080 fullscreen
[opengl.txt](https://github.com/pioneerspacesim/pioneer/files/4242075/opengl.txt)
[imgui.ini.gz](https://github.com/pioneerspacesim/pioneer/files/4242080/imgui.ini.gz)

NB: I also get the same crashes on Debian 10.2 with Gnome on XWayland

**Comment by Web-eWorks** [comment]
This is probably a memory corruption bug, which is going to require valgrinding the code with a fine toothed comb. Something we're doing is overflowing the imgui internal book-keeping related to mouse cursors.

Extremely likely to be related to #4762, and it may be as simple as us failing to reinitialize the cursors when the UI toggles on and off.

**Comment by cwyss** [comment]
I think I have found a fix both for this and for #4762. The fix, if it really works out, is indeed related to imgui mouse handling only and is as simple as dropping three lines of code.
Will do some more investigation and then report / put this in a PR, probably later this week.
